ocaml/typing/typemod.ml

```diff
@@ -2807,15 +2807,30 @@ and type_structure ?(toplevel = None) funct_body anchor env sstr =
                                    Toplevel_nonvalue (Ident.name id,sort)))
                 )
                 modes;
-              let (first_loc, _, _) = List.hd modes in
+              let zero_alloc =
+                (* We only allow "Check" attributes in signatures.  Here we
+                   convert "Assume"s in structures to the equivalent "Check" for
+                   the signature. *)
+                let open Builtin_attributes in
+                match[@warning "+9"] modes with
+                | [(_, _, _, (Default_check | Ignore_assert_all _))] ->
+                  Default_check
+                | [(_, _, _, (Check _ as zero_alloc))] -> zero_alloc
+                | [(_, _, _, Assume { property; strict; arity; loc;
+                                      never_returns_normally = _ })] ->
+                  Check { strict; property; arity; loc; opt = false }
+                | _ -> Default_check
+              in
+              let (first_loc, _, _, _) = List.hd modes in
```

Comment:
I am struggling to review this because I don't know what it means for `modes` to contain more than one element. See my other comment about multiple bindings in `let_bound_idents_with_modes_sorts_and_checks`.

Reply:
Thanks. I suggested a simplification in the other thread.